test/jdk/javax/swing/text/html/HTMLDocument/HTMLTextDecoration.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.Dimension;
+import java.awt.Graphics;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+
+import javax.imageio.ImageIO;
+import javax.swing.JEditorPane;
+import javax.swing.text.View;
+import javax.swing.text.html.CSS;
+
+/*
+ * @test
+ * @bug 8323801 8326734
+ * @summary Tests different combination of 'underline' and 'line-through';
+ *          the text should render with both 'underline' and 'line-through'.
+ * @run main HTMLTextDecoration
+ */
+public final class HTMLTextDecoration {
+    private static final String HTML = """
+            <!DOCTYPE html>
+            <html lang="en">
+            <head>
+                <meta charset="UTF-8">
+                <title>underline + line-through text</title>
+                <style>
+                    .underline   { text-decoration: underline }
+                    .lineThrough { text-decoration: line-through }
+                </style>
+            </head>
+            <body>
+            <p><u><span style='text-decoration: line-through'>underline + line-through?</span></u></p>
+            <p><s><span style='text-decoration: underline'>underline + line-through?</span></s></p>
+            <p><strike><span style='text-decoration: underline'>underline + line-through?</span></strike></p>
+
+            <p><span style='text-decoration: line-through'><u>underline + line-through?</u></span></p>
+            <p><span style='text-decoration: underline'><s>underline + line-through?</s></span></p>
+            <p><span style='text-decoration: underline'><strike>underline + line-through?</strike></span></p>
+
```

Comment:
Suppose there's this HTML
<p><s><span style='text-decoration: line-through'>underline + line-through?</span></s></p>


ie a strike through is specified in both ways. Does the merge code handle that ? I think it probably does but
adding this case to the test might be a good idea.

Reply:
Because you didn't add the backticks <code>`</code> around your sample, it's interpreted as HTML, and I can't really see it.

If the both tags have the same value for the `text-decoration` property, it works without the fix. I'm sure it works with the fix, however, in some cases the value of the property may be `line-through,line-through`.

I'll add another test if you think such a scenario is worth verifying too.

Reply:
I added the backticks so it now shows the source but all I did was "s/<u>/s/"

Yes, I think it might be wise to add such a test scenario.

Reply:
Added `HTMLUnderlineOnly.java` and `HTMLStrikeOnly.java` which test different combinations of setting `underline`  and `line-through` values for the `text-decoration` CSS property.

These two tests pass with and **without** the fix. This means neither is a regression test for [JDK-8326734](https://bugs.openjdk.org/browse/JDK-8326734). Shall I remove `8326734` from `@bug` tag from the tests?

Reply:
Up to you. Since they were developed in fixing this bug I don't have a problem with leaving it.